### DISCUSSION TITLE
Display

### DISCUSSION THREAD

**Discussion by souravlal** [marcel-licence/esp32_midi_sampler#38] [Q&A] [answered]
@marcel-licence Can you give me the link for the display you are using?

**Comment by marcel-licence** [comment]
Geekcreit 0.96 Inch 7Pin HD Color IPS Screen TFT LCD Display SPI ST7735 Module
---
Feature:
 
Size: 0.96 Inch
Material: New IPS color screen, the color is more vivid than ordinary TFT LCD screen
Resolution: 80*160 display, direction can adjust horizontal screen vertical screen
Control chip: ST7735
Display area: 10.8 x 21.696mm
Physical size: 24 x 30mm
Interface type: SPI interface
Number of pins: 7
Display color: 65K full color
Working temperature: -20~70°C
Working voltage: 3.3V
Net weight: 5g
 
Interface definition:
 
1> GND (Power ground)
2> VCC (Power +3.3)
3> SCL (SPI clock line)
4> SDA (SPI data line)
5> RES (LCD reset, LCD needs to be reset once after power-on)
6> DC (SPI data/command selection pin)
7> BLK (LCD backlight control, default can be left floating, low level to turn off the backlight)

**Reply by marcel-licence** [reply]
![grafik](https://user-images.githubusercontent.com/65245426/127393471-69cbf0ed-8ac8-4494-a7a4-c28ac04952e0.png)

**Reply by souravlal** [reply]
Can you give me the link of the product?

**Reply by souravlal** [reply]
Can you please also give me the connection of the display to the ESP32 Audio-kit?

**Comment by souravlal** [comment]
@marcel-licence Can I use this display instead? with the same code?

https://robu.in/product/blue-1-8-inch-st7735-tft-lcd-module-with-4-io-128160/

**Comment by souravlal** [comment]
Please somebody help me with this👆

**Reply by marcel-licence** [reply]
I would like to kindly ask you to avoid the reminders. I think you will get a good support. Developing, finding components etc. requires a lot of time. The connections are defined in config.h. Yes you can also use another ST7735 compatible display. I don't know how good the adafruit library will work with it. The display I am using was not bug free supported by adafruit.

**Reply by marcel-licence** [reply]
#define TFT_MOSI    0
#define TFT_SCLK    19
#define TFT_CS        23
#define TFT_RST       -1 // 5 // Or set to -1 and connect to Arduino RESET pin -> RST on Audio Kit
#define TFT_DC         21
#define TFT_BLK_PIN 14  // do not connect it would cause the Audio Kit to get stuck -> leave this disconnected

**Reply by souravlal** [reply]
@marcel-licence Can you give me a close picture of your display?
The displays I have found doesn't seems to have mosi, miso, sclk and cs pins.
But its specs say its spi

**Reply by marcel-licence** [reply]
https://www.google.com/search?q=connect+st7735+0.96+display
Its confusing that you will find SDA and SCL but this is used as MOSI and SCK

**Comment by Slider2732** [comment, marked as answer]
As long as the ad states it can be driven with the ST7735 library, the pin names can be worked out in a couple of minutes.
You may get SDA and SCL or other names. As long as VCC and GND are correct there's little to go wrong. 
On my 0.96" the backlight (end pin marked BLK), I've soldered a 1K resistor to VCC.
I know I just looked at the code and guessed the pins. Some were correct by name, others not. RES equaled RST, easy enough. 
Nothing breaks with the wrong data lines connected  :)
